test/maxtree.jl

```diff
@@ -0,0 +1,237 @@
+@testset "connectivity neighborhoods" begin
+    @test @inferred(ImageMorphology.neighbor_cartesian_offsets(CartesianIndex{1}, 1)) ==
+          ImageMorphology.neighbor_cartesian_offsets(CartesianIndex{1}, 2) ==
+          [CartesianIndex{1}(-1), CartesianIndex{1}(1)]
+    conn2d_4 = ImageMorphology.neighbor_cartesian_offsets(CartesianIndex{2}, 1)
+    @test conn2d_4 ==
+        [CartesianIndex{2}(0, -1), CartesianIndex{2}(-1, 0),
+         CartesianIndex{2}(1, 0), CartesianIndex{2}(0, 1)]
+    @test @inferred(ImageMorphology.neighbor_cartesian_offsets(CartesianIndex{2}, 2)) ==
+          [CartesianIndex{2}(-1, -1), CartesianIndex{2}(0, -1), CartesianIndex{2}(1, -1),
+           CartesianIndex{2}(-1, 0), CartesianIndex{2}(1, 0),
+           CartesianIndex{2}(-1, 1), CartesianIndex{2}(0, 1), CartesianIndex{2}(1, 1)]
+    @test ImageMorphology.linear_offsets(conn2d_4, fill(0, (1, 1))) == [-1, -1, 1, 1]
+    @test ImageMorphology.linear_offsets(conn2d_4, fill(0, (2, 2))) == [-2, -1, 1, 2]
+    @test @inferred(ImageMorphology.linear_offsets(conn2d_4, fill(0, (3, 3)))) == [-3, -1, 1, 3]
+    @test @inferred(ImageMorphology.linear_offsets(conn2d_4, fill(0, (7, 9)))) == [-7, -1, 1, 7]
+end
+
+const CI2 = CartesianIndex{2}
+
+bbox2d(x1, y1, x2, y2) = (CI2(x1, y1), CI2(x2, y2))
+
+@testset "MaxTree construction" begin
```

Comment:
Please don't forget to add a testset for offset arrays. For example,

```julia
A0 = OffsetArray(A, -1, -1)
mtree_0 = MaxTree(A0, connectivity=1, rev=false)

# the current design is expected to fail here
@test collect(mtree_0.parentindices) == mtree.parentindices
```

If you're not going to support it in this PR, make sure `MaxTree(A0, connectivity=1, rev=false)` errors.